### PR TITLE
Add IAM role clean to cleaner.

### PR DESCRIPTION
**Pull request by jefchien**
**Description:** Cleans IAM roles starting with `terraform`, `aoc-eks-assume-role`, or `fargate-profile-role`.

**Testing:** `export AWS_SDK_LOAD_CONFIG=true && go run tools/workflow/clean_aws_resources.go` with other functions removed.

### CODE REVIEW

File: tools/workflow/clean_aws_resources.go [hunk 1]
```diff
@@ -322,3 +328,67 @@ func destroyLoadBalancerResource() {
...
+		}
+
+		for _, role := range lro.Roles {
+			var hasPrefix bool
+			for _, prefix := range roleNamePrefixes {
+				if hasPrefix = strings.HasPrefix(*role.RoleName, prefix); hasPrefix {
+					break
+				}
+			}
+			if hasPrefix && time.Now().UTC().Add(pastDayDeleteCalculation).After(*role.CreateDate) {
```
Comment (sethAmazon) [inline tools/workflow/clean_aws_resources.go:358]: We may want to also have it delete by date since last used

Comment (jefchien) [reply tools/workflow/clean_aws_resources.go:358]: For some reason this is always nil, but I can add it in on the off chance that it's available.

```go
if role.RoleLastUsed != nil {
  log.Printf("Last used %s %v", *role.RoleName, *role.RoleLastUsed.LastUsedDate)
}
```

Comment (khanhntd) [reply tools/workflow/clean_aws_resources.go:358]: Prefer last used date instead of create date.

Comment (jefchien) [reply tools/workflow/clean_aws_resources.go]: It's nil for all of the roles in our account. I've added it to the conditional in case it's ever not nil.

Comment (sethAmazon) [reply tools/workflow/clean_aws_resources.go]: nil I think means not used

**Review by sethAmazon — APPROVED** [review]

File: tools/workflow/clean_aws_resources.go (hunk 1 above)
Comment (jefchien) [reply tools/workflow/clean_aws_resources.go]: It seems like a bug in the AWS SDK https://github.com/aws/aws-sdk/issues/24

**Pull request merged**